src/components/SideBar.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import Logo from '../molecules/Logo';
+import styled from 'styled-components';
+import MainActionButton from './MainActionButton';
+import { Semester, Course } from '../api';
+export interface SideBarInterface {
+	currentSemester: Semester;
+}
+const SideBarContainer = styled.div`
+	background-color: #4961e1;
+	height: 100%;
+	width: 100%;
+	justify-content: center;
+	text-align: center;
+	font-weight: bold;
+`;
+const ListOfCoursesContainer = styled.ol`
+	font-size: large;
+	padding: 0;
+`;
+const CourseItem = styled.li`
+	color: #e5f2fc;
+	cursor: pointer;
+	padding-top: 1em;
+	padding-bottom: 1em;
+	list-style: none;
+	&:hover {
+		background-color: #f2f2f2;
+		color: #002366;
+		list-style: none;
+	}
+`;
+const SideBarImage = styled(Logo)`
+	background-color: #4961e1;
+`;
+const AddCourseButtonContainer = styled.div`
+	margin: 2em 1em;
+	font-size: small;
+`;
+
+function AddCourse() {
+	console.log('course added');
+}
+
+function CourseList({ courses }: { courses: Course[] }) {
+	return (
+		<ListOfCoursesContainer>
+			{courses.map((item, index) => {
+				return <CourseItem key={index}>{item.name}</CourseItem>;
```

Comment:
So I feel like I have used index as the key before, and honestly it might be fine in this scenario, but the reason react uses a key in a list is so that it can be efficient in list rendering. I think we should use maybe the `item.name` as the key. It may prove more efficient

Reply:
https://reactjs.org/docs/lists-and-keys.html#keys

Here they mention that you should use a unique identifier. Since I can't imagine someone being in the same course twice, I assume that using the `item.name` as the key should be great! 

This article goes into detail about lists and keys. Read it if you wish!

Reply:
On it! :)